memgpt/utils.py

```diff
@@ -88,11 +88,25 @@ def parse_json(string):
         raise e
 
 
-def list_agent_config_files():
+def list_agent_config_files(sort="last_modified"):
     """List all agent config files, ignoring dotfiles."""
-    files = os.listdir(os.path.join(MEMGPT_DIR, "agents"))
-    #  remove dotfiles like .DS_Store
-    return [file for file in files if not file.startswith(".")]
+    agent_dir = os.path.join(MEMGPT_DIR, "agents")
+    files = os.listdir(agent_dir)
+
+    # Remove dotfiles like .DS_Store
+    files = [file for file in files if not file.startswith(".")]
+
+    # Remove anything that's not a directory
+    files = [file for file in files if os.path.isdir(os.path.join(agent_dir, file))]

```

Comment:
@sarahwooders should we add an additional check to make sure that a save file exists? I remember having some issues with agents that never had saves written out still showing up as "loadable", but then throwing a runtime error if you tried to run them.